### ISSUE TITLE
isConnected was still true when broken pipe or close connection in channel->wait()

### ISSUE THREAD

**Issue by namjeongho** [php-amqplib/php-amqplib#389]
Reconnecting is not possible at a broken pipe runtime exception in channel->wait()
because isConnected was still true when broken pipe or close connection in channel->wait()
This is like issue #259.

**Comment by postalservice14** [comment]
What version are you running?

Do you know what was calling channel->wait() when you got the broken pipe?  (Stack trace would be helpful)  The reason I ask, is because ideally, the thing calling channel->wait will react to the exception and set isConnected to false.

**Comment by namjeongho** [comment]
I use php-amqplib 2.6.3 and rabbitmq-server 3.4.2 in CentOS 6.

I tested the situation that rabbitmq-server process is killed during worker process run.
This is main loop in my test code.  
```
while (count($channel->callbacks)) {
    try {
        $channel->wait();
    } catch (Exception $e) {
        echo $e->__toString();
        if ($connection->isConnected() == false) {
            connect();
        }
    }
}
```
If I shutdown rabbitmq-server gracefully, AMQPProtocolConnectionException was thrown
and $connection->isConnected() return true.
```/etc/init.d/rabbitmq-server stop```

But if I kill rabbitmq-server process **by force** not gracefully, AMQPRuntimeException was thrown, 
and $connection->isConnected() return **false**.
```kill {rabbitmq-server process pid}```

At this moment, $e->__toString() print below.

```
exception 'PhpAmqpLib\Exception\AMQPRuntimeException' with message 'Broken pipe or closed connection' in /mnt/project/mq/lib/vendor/php-amqplib/php-amqplib/PhpAmqpLib/Wire/IO/StreamIO.php:207
Stack trace:
#0 /mnt/project/mq/lib/vendor/php-amqplib/php-amqplib/PhpAmqpLib/Wire/AMQPReader.php(149): PhpAmqpLib\Wire\IO\StreamIO->read(7)
#1 /mnt/project/mq/lib/vendor/php-amqplib/php-amqplib/PhpAmqpLib/Wire/AMQPReader.php(106): PhpAmqpLib\Wire\AMQPReader->rawread(7)
#2 /mnt/project/mq/lib/vendor/php-amqplib/php-amqplib/PhpAmqpLib/Connection/AbstractConnection.php(508): PhpAmqpLib\Wire\AMQPReader->read(7)
#3 /mnt/project/mq/lib/vendor/php-amqplib/php-amqplib/PhpAmqpLib/Connection/AbstractConnection.php(555): PhpAmqpLib\Connection\AbstractConnection->wait_frame(0)
#4 /mnt/project/mq/lib/vendor/php-amqplib/php-amqplib/PhpAmqpLib/Channel/AbstractChannel.php(217): PhpAmqpLib\Connection\AbstractConnection->wait_channel(1, 0)
#5 /mnt/project/mq/lib/vendor/php-amqplib/php-amqplib/PhpAmqpLib/Channel/AbstractChannel.php(328): PhpAmqpLib\Channel\AbstractChannel->next_frame(0)
#6 /mnt/project/mq/main.php(11): PhpAmqpLib\Channel\AbstractChannel->wait()
```

**Comment by postalservice14** [comment]
Yeah, channel->wait() throwing an exception itself won't close the connection, but the thing calling it should.

**Comment by namjeongho** [comment]
Sorry, I modified my previous comment.
In summary, I expected that $connection->isConnected() return false when Rabbitmq server stop in any case and Exception occurred.

**Comment by danielzzz** [comment]
I have the same problem. It happens when the task takes longer to complete than the timeout/heartbeat value. The server drops the connection, but $connection->isConnected() still returns true and it hangs on $channel->wait();

**Comment by pfbernardo** [comment]
@namjeongho did you found a solution for this? Having the same problem.

**Comment by erick03** [comment]
@danielzzz did you figure out how to solve that problem?

**Comment by danielzzz** [comment]
@erick03 actually I don't remember %) it's a project I haven't touched in months. 
but looking in the code, I just quit when there is an exception (and probably start the worker again using a process manager).

this is my main loop: ( l($txt) is just a simple log function
```php
// -------------- main loop -----------------------
echo ' [*] Waiting for messages. To exit press CTRL+C', "\n";
while (count($channel->callbacks) && $channel->is_open) {
    sleep(10);
    try {
        $channel->wait();
    } catch (PhpAmqpLib\Exception\AMQPTimeoutException $e) {
        $r = "idle timeout, exiting...";
        close($r);
        l($r);
        exit();
    } catch (PhpAmqpLib\Exception\AMQPRuntimeException $e) {
        $r = "caught runtime exception, exiting...";
        close($r);
        l($r);
        exit();
    } catch (Exception $e) {
        $r = $e->getMessage();
        close($r);
        l($r);
        exit();
    }
}
```

**Comment by lukebakken** [comment]
Please test this scenario with the upcoming `2.9.0-rc1` candidate. I will assume this is fixed if I don't hear back within a week or so.

**Comment by lukebakken** [comment]
Assuming resolved by `2.9.0`. If not, please comment.
